### ISSUE TITLE
Optionals arguments not correctly interpreted

### ISSUE THREAD

**Issue by mbl-35** [mbl-35/wslctl#1]
wslctl: v1.0.0
wins10

When using with powershell inerpreter, no pb:

```powershell
PS> powershell .\src\wslctl.ps1 --help
Usage:
   wslctl COMMAND [ARG...]
   wslctl [ --help | --version ]
...
```
But with the generated executable, the command line does not pass '--' but '-'  :

```powershell
PS> wslctl --help
Error: Command '-help' is not defined
PS > wslctl ---help
Usage:
   wslctl COMMAND [ARG...]
   wslctl [ --help | --version ]
...
```

Same from MSDos console:

```powershell
C:\Users>wslctl --help
Error: Command '-help' is not defined

C:\Users>wslctl ---help

Usage:
   wslctl COMMAND [ARG...]
   wslctl [ --help | --version ]
```

**Comment by mbl-35** [comment]
Issue from ps2exe: https://github.com/MScholtes/PS2EXE/issues/54#issuecomment-897698775
If I want to keep UN*x behavior, I need to change my parsing parameter method (but no powershell names...)

**Comment by mbl-35** [comment]
v1.0.1

Still have error with .exe
```powershell
--force : toot many arguments
--no-user: invalid parameter
--v1 invalid parameter
```

**Comment by mbl-35** [comment]
Simple test.ps1
```powershell
$args.count
$args | ConvertTo-Json
```
ps1 tests:
```powershell
PS > .\test.ps1 arg1 arg2 arg3
3
[
    "arg1",
    "arg2",
    "arg3"
PS > .\test.ps1 arg1 arg2 arg3 --toto
4
[
    "arg1",
    "arg2",
    "arg3",
    "--toto"
]
```

Create binary with `Invoke-ps2exe -inputFile test.ps1 -outputFile test.exe -x64` and do same tests:

```powershell
PS > .\test.exe arg1 arg2 arg3 --toto
5
[
    "arg1",
    "arg2",
    "arg3",
    "-toto",
    true
]
PS > .\test.exe arg1 arg2 arg3 ---toto
5
[
    "arg1",
    "arg2",
    "arg3",
    "--toto",
    true
]
PS > .\test.exe arg1 arg2 arg3 -toto
5
[
    "arg1",
    "arg2",
    "arg3",
    "-toto",
    true
]
PS > .\test.exe arg1 arg2 arg3 -toto --toto
7
[
    "arg1",
    "arg2",
    "arg3",
    "-toto",
    true,
    "-toto",
    true
]
```

Seems that ps2exe add boolean when parameter starts with '-'.

**Solution**: Filter args
New test.ps1
```powershell
$args = $args | Where {$_ -is [String]}  # Filter
$args.count
$args | ConvertTo-Json
```
Produce same result calling the ps1 file.

**Results** exe recompiled with filter

```Powershell
PS > .\test.exe arg1 arg2 arg3 --toto
4
[
    "arg1",
    "arg2",
    "arg3",
    "-toto"
]
PS > .\test.exe arg1 arg2 arg3 -toto
4
[
    "arg1",
    "arg2",
    "arg3",
    "-toto"
]
PS > .\test.exe arg1 arg2 arg3 ---toto
4
[
    "arg1",
    "arg2",
    "arg3",
    "--toto"
]
PS > .\test.exe arg1 arg2 arg3 -toto --toto
5
[
    "arg1",
    "arg2",
    "arg3",
    "-toto",
    "-toto"
]
```

**Comment by mbl-35** [comment]
Resolved by release 1.0.2.
